NumericOutcome.md

```diff
@@ -0,0 +1,437 @@
+
+## Contract Execution Transaction Compression
+
+Anytime there is a range of numeric outcomes `[start, end]` which result in the same payouts for all parties,
+then a compression function can be run to reduce the number of CETs to be `O(log(L))` instead of `L` where 
+`L = end - start + 1` is the length of the interval being compressed.
+
+Most contracts are expected to be concerned with some subset of the total possible domain and every
+outcome before or after that range will result in some constant maximal or minimal payout.
+This means that compression will drastically reduce the number of CETs to be of the order of the size
+of the probable domain, with further optimizations available when parties are willing to do some rounding.
+
+The compression algorithm takes as input a range `[start, end]`, a base `B`, and the number of digits
+`n` (being signed by the oracle) and returns an array of arrays of integers (which will all be in the range `[0, B-1]`).
+An array of integers corresponds to a single event equal to the concatenation of these integers (interpreted in base `B`).
+
+### Example
+
+Before specifying the algorithm, let us run through a general example.
+
+Consider the range `[(prefix)wxyz, (prefix)WXYZ]` where `prefix` is some string of digits in base `B` which
+`start` and `end` share and `w, x, y, and z` are the unique digits of `start` in base `B` while `W, X, Y, and Z`
+are the unique digits of `end` in base `B`.
+
+To cover all cases while looking at as few digits as possible in this (general) range we need only consider
+`(prefix)wxyz`, `(prefix)WXYZ` and the following cases:
+
+```
+wxy(z+1), wxy(z+2), ..., wxy(B-1),
+wx(y+1)_, wx(y+2)_, ..., wx(B-1)_,
+w(x+1)__, w(x+2)__, ..., w(B-1)__,
+
+(w+1)___, (w+2)___, ..., (W-1)___,
+
+W0__, W1__, ..., W(X-1)__,
+WX0_, WX1_, ..., WX(Y-1)_,
+WXY0, WXY1, ..., WXY(Z-1)
+```
+
+where `_` refers to an ignored digit (an omission from the array of integers) and all of these cases have the `prefix`.
+I refer to the first three rows as the **front groupings** the fourth row as the **middle grouping** and the last three rows
+as the **back groupings**.
+
+Notice that the patterns for the front and back groupings are nearly identical, and that in total the number of CETs that
+will be required to cover the range will be equal to the sum of the unique digits of `end` plus the sum of `B-1` minus the
+unique digits of `start`.
+This means that the number of CETs required to cover a range of length `L` will be `O(B*log_B(L))` because `log_B(L)`
+corresponds to the number of unique digits between the start and end of the range and for each unique digit a row is
+generated in both the front and back groupings of length at most `B-1 ` which corresponds to the coefficient in the order bound.
+This counting shows us that base 2 is the optimal base to be using in general cases as it will outperform all larger bases
+in both large and small ranges in general.
+
```

Comment:
The number of CETs can be further reduced by selecting  (B^ n-1) or  (B ^ n) for `start` and ʻend`.

Reply:
So while this is true, I'm not sure that it is particularly practical?

Reply:
This idea is valid in both decimal and binary numbers, so I think it is practical. As a more practical description, it may be better to add the following to the #optimizations chapter.

Choosing a value closer to (B ^ n) for `start` and `end` will reduce the number of CETs.